configure.ac

```
@@ -2252,6 +2252,20 @@ AC_CHECK_FUNCS(waitpid)
 AC_CHECK_FUNCS(__cospi)
 AC_CHECK_FUNCS(__sinpi)
 
+AC_ARG_ENABLE(macos-localtime-cache,
+	AS_HELP_STRING([--enable-macos-localtime-cache],
+		       [enable localtime cache optimization for macOS to improve forked process performance]),
+	[macos_localtime_cache=$enableval], [macos_localtime_cache=yes])
+AS_IF([test "$macos_localtime_cache" = yes], [
+    AS_CASE(["$target_os"],
+        [darwin*], [
+            AC_DEFINE(ENABLE_MACOS_LOCALTIME_CACHE, 1, [Enable macOS localtime cache optimization])
+            AC_MSG_NOTICE([macOS localtime cache optimization enabled])
+        ],
+        [AC_MSG_WARN([--enable-macos-localtime-cache is only supported on macOS, ignoring])]
+    )
```

It is not nice this warning is shown unless this option is explicitly disabled on other platforms.
https://github.com/ruby/ruby/actions/runs/18602388839/job/53043740969#step:6:531

```suggestion
	[AS_CASE(["$target_os"],
	    [darwin*], [macos_localtime_cache=$enableval],
	    [*], [AC_MSG_WARN([--enable-macos-localtime-cache is only supported on macOS, ignoring])]
	)],
	[AS_CASE(["$target_os"], [darwin*], [macos_localtime_cache=yes], [macos_localtime_cache=no])])
AS_IF([test "$macos_localtime_cache" = yes], [
    AC_DEFINE(ENABLE_MACOS_LOCALTIME_CACHE, 1, [Enable macOS localtime cache optimization])
    AC_MSG_NOTICE([macOS localtime cache optimization enabled])
```